frontend/src/Frontend/Examples/ScreenKeyboard/Main.hs

```diff
@@ -50,8 +52,24 @@ inputW buttonE = do
     let
       html = _inputElement_raw input     -- html element
       cur = current $ value input        -- actual string
-    input <- inputElement $ def
-      & inputElementConfig_setValue .~ fmap snd newStringE
+      cfg = def & inputElementConfig_setValue .~ fmap snd newStringE
+
+    input <- inputElement cfg
+    -- Use this to trigger an event *after* the value has been updated.
+    valueChangedBySetValue :: Event t String <-
+      case _inputElementConfig_setValue cfg of
+        Nothing -> return never
+        Just eSetValue -> requestDomAction $ getValue html <$ eSetValue
```

Comment:
An issue I kept encountering was the value changing here *after* all the other events within the frame were fired. This seems right - we want all events to have the same view of the input before anything updates. But it does introduce a problem where updating this value places the cursor at the end of the input meaning I need to update the cursor's position in the following frame. To get around this, I copied some internal code from [Reflex.Dom.Builder.Immediate](https://hackage.haskell.org/package/reflex-dom-core-0.7.0.1/docs/Reflex-Dom-Builder-Immediate.html) to trigger a new event but this feels pretty hacky. Is there a better way to trigger an event after the current frame?